project-addons/custom_account/report/account_invoice_report.py

```diff
@@ -41,14 +41,16 @@ def _sub_select(self):
         select_str += ', ai.payment_mode_id,' \
                       ' ai.number ' \
                       ', sum(sol.product_uom_qty * sol.price_unit * (100.0-sol.discount) ' \
-                      '/ 100.0) - sum(sol.purchase_price*sol.product_uom_qty) as benefit'
+                      '/ 100.0) - sum(coalesce(ip.value_float, 0)*sol.product_uom_qty) as benefit'
 
         return select_str
 
     def _from(self):
         from_str = super(account_invoice_report, self)._from()
         from_str += ' LEFT JOIN sale_order_line_invoice_rel solir ON solir.invoice_id = ail.id ' \
-                    ' LEFT JOIN sale_order_line sol ON sol.id = solir.order_line_id '
+                    ' LEFT JOIN sale_order_line sol ON sol.id = solir.order_line_id ' \
+                    " LEFT JOIN ir_property ip ON (ip.name= 'standard_price' " \
+                    " AND ip.res_id=CONCAT('product.template,',pr.product_tmpl_id))"
```

Comment:
No entiendo lo que pretendéis aquí, de esta forma siempre está obteniendo el último precio de coste del producto, en lugar del precio de coste que tuvo en su momento, es decir que si hace 1 año obteníais un 30% por un producto y hoy sólo obtenéis un 10% estáis cambiando el beneficio de las ventas del año pasado.